signalingserver.go

```diff
@@ -14,23 +14,161 @@
 
 package main
 
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+)
+
 // The ForwardingSignalingServer implements the SignalingServer interface by
 // communicating with the host orchestrator in the appropriate instance and
 // forwarding all requests to it.
-type ForwardingSignalingServer struct{}
+type ForwardingSignalingServer struct {
+	instanceManager InstanceManager
+}
 
-func NewForwardingSignalingServer() *ForwardingSignalingServer {
-	return &ForwardingSignalingServer{}
+func NewForwardingSignalingServer(im InstanceManager) *ForwardingSignalingServer {
+	return &ForwardingSignalingServer{im}
 }
 
-func (s *ForwardingSignalingServer) NewConnection(msg NewConnMsg) NewConnReply {
-	return NewConnReply{ConnId: "id", DeviceInfo: "device info"}
+func (s *ForwardingSignalingServer) NewConnection(msg NewConnMsg) (SServerResponse, error) {
+	dev, err := s.instanceManager.DeviceFromId(msg.DeviceId)
+	if err != nil {
+		return SServerResponse{StatusCode: http.StatusInternalServerError}, err
+	}
+	var resErr ErrorMsg
+	var reply NewConnReply
+	status, err := POSTRequest(hostURL(dev.Addr, "/polled_connections", ""), NewConnMsg{dev.LocalId}, &reply, &resErr)
+	if err != nil {
+		return SServerResponse{StatusCode: status}, err
```

Comment:
Is it possible to use [error wrapping](https://go.dev/doc/go1.13#error_wrapping) to attach more information to the errors returned from these functions? It looks like of the error handling here logs the extra information and then returns the original error. Augmenting the original error would allow constructing a single error object with all the relevant context. See also the [go 1.13 errors blog post](https://go.dev/blog/go1.13-errors).

Reply:
Yes, this is temporary. I need to figure out what's the best way to configure the app. That configuration will include the port used by the host orchestrator, the protocol and possible some auth metadata.

Reply:
Done